src/dotnet-uninstall/Shared/Filterers/AllBelowOptionFilterer.cs

```diff
@@ -1,17 +1,29 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
 using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo.Versioning;
+using Microsoft.DotNet.Tools.Uninstall.Shared.Exceptions;
 
 namespace Microsoft.DotNet.Tools.Uninstall.Shared.Filterers
 {
     internal class AllBelowOptionFilterer : ArgFilterer<string>
     {
         public override IEnumerable<Bundle<TBundleVersion>> Filter<TBundleVersion>(string argValue, IEnumerable<Bundle<TBundleVersion>> bundles)
         {
-            var specifiedVersion = BundleVersion.FromInput<TBundleVersion>(argValue) as TBundleVersion;
+            try
+            {
+                var version = BundleVersion.FromInput<TBundleVersion>(argValue) as TBundleVersion;
 
-            return bundles
-                .Where(bundle => bundle.Version.CompareTo(specifiedVersion) < 0);
+                return bundles
+                    .Where(bundle => bundle.Version.CompareTo(version) < 0);
+            }
+            catch (InvalidInputVersionException)
```

Comment:
This is exception as control flow. Unless it is runtime exception, like networking is down, once throw, should error to the top. If you can handle it like the following, use tester-doer. (although tester-doer is not thread safe)

Reply:
C# doesn't distinguish runtime exceptions and checked exceptions, so essentially every exception is a runtime exception (if what you meant by "runtime exception" is the `RuntimeException` in Java).

The main issue for using exceptions as control flows is performance (see https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exceptions-and-performance). However this part will be run _at most once_ for each execution of the command line tool. Hence I don't think it's a big issue for us.

I could change it to the tester-doer though.

Reply:
Done.